cli/azd/pkg/azure/arm_template.go

```diff
@@ -24,11 +26,11 @@ type RawArmTemplate = json.RawMessage
 // at https://learn.microsoft.com/azure/azure-resource-manager/templates/syntax, but only exposes portions of the
 // object that azd cares about.
 type ArmTemplate struct {
-	Schema         string                          `json:"$schema"`
-	ContentVersion string                          `json:"contentVersion"`
-	Parameters     ArmTemplateParameterDefinitions `json:"parameters"`
-	Outputs        ArmTemplateOutputs              `json:"outputs"`
-	Definitions    ArmTemplateParameterDefinitions `json:"definitions"`
+	Schema         string                                               `json:"$schema"`
+	ContentVersion string                                               `json:"contentVersion"`
+	Parameters     custommaps.WithOrder[ArmTemplateParameterDefinition] `json:"parameters"`
```

Comment:
Must we do this?  I think it is much easier to reason about things if we don't bake ordering in at this layer.  What pushed you towards needing to retain the order here?

Reply:
When we prompt for parameters, we are currently first sorting the parameters to make the prompting deterministic.
I updated the type to use the `withOrder[T]` and removed the sorting.

It is not a required thing for the secrets-stuff..  I think I can move it to a different PR :)

Reply:
Moved those changes to: https://github.com/Azure/azure-dev/pull/3724